### PR TITLE
Add initial release-process.md

### PR DESCRIPTION
**Pull request by gotmax23**
Relates: https://github.com/ansible-community/ansible-build-data/issues/221

### CODE REVIEW

**Comment by gotmax23** [comment]
@anweshadas (and anyone else who has done the releases in the past that feels like chiming in) , I'd really appreciate your opinion here :smile:. Am I missing anything important? Is there anything that you do differently? Is there anything that's incorrect?

Also I'd appreciate more information about the release announcements (step `#8`). I'm not sure where the the Matrix template comes from.

**Comment by felixfontein** [comment]
CC @rooftopcellist

**Comment by anweshadas** [comment]
I am working on this. Will be sharing this by this week.

File: docs/release-process.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+## Preamble
+
+This document describes the ansible community package release process.
+
+> **Note**
+>
+> Throughout this page, placeholder values in code blocks are formatted as
+> `{PLACEHOLDER}` where `PLACEHOLDER` describes the value to fill in.
+
```
[the remark this answers is not in the document]
Comment (gotmax23) [reply docs/release-process.md:11]: Why does the release need to be done inside of a container?

File: docs/release-process.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+   switch to the ansible checkout,
+   commit and push the changes,
+   and then submit a PR as you normally would.
+   You can use `Add Ansible community {VERSION} porting guide` as the commit message.
+
+6. Once the ansible-build-data PR has been merged,
+   publish the build artifacts to PyPI.
+   From the antsibull repository root, run
+
+   ```
```
[the remark this answers is not in the document]
Comment (gotmax23) [reply docs/release-process.md:117]: Where does this key come from? Do all release managers have a copy of it?

**Comment by gotmax23** [comment]
Thanks for the review @anweshadas! I left some follow up comments.

File: docs/release-process.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (anweshadas) [reply docs/release-process.md:117]: Currently it is with the Red Hat.

File: docs/release-process.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (anweshadas) [reply docs/release-process.md:11]: For having a clean build environment, which is  also better for standard security ideas.

[the remark this answers is not in the document]
Comment (anweshadas) [reply docs/release-process.md:11]: We created a template for the announcement templates. We need to figure out a place to publish it. @gotmax23 can you suggest something?

[the remark this answers is not in the document]
Comment (felixfontein) [reply docs/release-process.md:11]: I would put the template(s) in this repository as well (maybe some subdirectory of `docs/`, like `docs/templates/`?). That way it's also easier for folks to suggest improvements to the templates (by creating a PR).

[the remark this answers is not in the document]
Comment (gotmax23) [reply docs/release-process.md:11]: I agree with @felixfontein that this repo is a good place to publish the announcement templates. I think a `docs/announcements` or maybe just a separate folder at the repo root would work.

[the remark this answers is not in the document]
Comment (gotmax23) [reply docs/release-process.md:11]: See the latest commit.

File: docs/release-process.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (gotmax23) [reply docs/release-process.md:130]: I think this is now a moot point. See https://blog.pypi.org/posts/2023-05-23-removing-pgp/.

/resolve

**Comment by gotmax23** [comment]
Triggering CI

File: docs/release-process.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+```
+
+
+## Set up repository clones
+
+First, you need to set up ansible-build-data and antsibull repository clones.
+This only needs to be done once.
+
+1. [Fork][abd-fork] the [ansible-build-data] repository.
+2. Checkout the antsibull and ansible/ansible (ansible-core) repositories
```
Comment (felixfontein) [inline docs/release-process.md:32]:
```suggestion
2. Checkout the antsibull and ansible/ansible-documentation repositories
```

File: docs/release-process.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+   and change into antsibull.
+
+   ```
+   git clone https://github.com/ansible/ansible
```
Comment (felixfontein) [inline docs/release-process.md:36]:
```suggestion
   git clone https://github.com/ansible/ansible-documentation
```

File: docs/release-process.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+   cd build/ansible-build-data
+   git switch -c release-{VERSION}
+   git add {MAJOR VERSION}/
+   git commit -m "Ansible {VERSION}: Dependencies, changelog and porting guide"
+   git push -u {USERNAME} release-{VERSION}
+   ```
+
+   Then, submit a pull request against ansible-build-data upstream.
+
+5. Submit a PR to ansible/ansible to add the new porting guide to the docsite.
```
Comment (felixfontein) [inline docs/release-process.md:113]:
```suggestion
5. Submit a PR to ansible/ansible-documentation to add the new porting guide to the docsite.
```

**Review by felixfontein — COMMENTED** [review]
The ansible docsite is now part of ansible/ansible-documentation.

File: docs/release-process.md (hunk 5 above)
Comment (felixfontein) [reply docs/release-process.md:113]: @oraNod @samccann is this ready for updating the release process accordingly?

File: docs/release-process.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+   Make sure you have the `main` branch checked out
+   and run `git pull` to update to the latest commit.
+
+2. Create a clean virtual environment for the release process.
+
+   ```
+   rm -rf release-venv
+   python3 -m venv release-venv
+   . ./release-venv/bin/activate
+   pip install -U pip
```
Comment (anweshadas) [inline docs/release-process.md:71]: `python3 -m pip install -U pip`

File: docs/release-process.md [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+   ```
+
+   Install the `antsibull`, `ansible-core`, and `twine` python packages,
+   as well as the community.general collection.
+
+   ```
+   pip install antsibull ansible-core twine
```
Comment (anweshadas) [inline docs/release-process.md:78]: `python3 -m pip install antsibull ansible-core twine`

**Review by anweshadas — CHANGES_REQUESTED** [review]
Apart from those few small changes everything looks good.

**Comment by anweshadas** [comment]
Created the PR https://github.com/ansible-community/ansible-build-data/pull/258 for the  announcement templates.

**Pull request merged**